TriblerGUI/widgets/trustdisplaypage.py

```diff
@@ -0,0 +1,160 @@
+            {
+                "focus_node": "xyz",
+                "neighbor_level: 1
+                "nodes": [{
+                    "public_key": "xyz",
+                    "page_rank": 5
+                }, ...],
+                "edges": [{
+                    "from": "xyz",
+                    "to": "xyz_n1",
+                    "amount_up": 100,
+                    "amount_down": 10,
+                    "ratio": 0.90909,
+                    "log_ratio": 0.66666
+                }, ...]
+            }
+        
+
+        :param information: the JSON dictionary passed on by the HTTP Request
+        """
+        focus_node = information["focus_node"]
+
+        grouped_edges = TrustDisplayPage.group_elements(information["edges"], "from")
+
+        # TODO: Do the procedure below for each of the nodes in the graph, whilst erasing duplicates.
+        focus_node_edges = TrustDisplayPage.get_combined_edges(grouped_edges, focus_node)
+
+        # TODO: Send the data to the display to display the data
+        return {"nodes": information["nodes"], "edges": focus_node_edges}
+
+    @staticmethod
+    def get_combined_edges(grouped_edges, node_name):
+        """
+        Combine the directed edges between the given node and other nodes to one edge per pair.
+
+        The attributes of the combined edges are calculated as follows:
+            - from: node_name
+            - to: to attribute from outgoing edge from node_name
+            - amount_up: amount from the outgoing edge from node_name
+            - amount_down: amount from the ingoing edge to node_name if any
+            - ratio: amount_up / (amount_up + amount_down)
+            - log_ratio: log(amount_up + 1) / (log(amount_up + 1) + log(amount_down + 1))
+
+        :param grouped_edges: the dictionary of edges, grouped by "from" attribute
+        :param node_name: the node name from which viewpoint each combine edge is created
+        :return: a list of combined edges with the described attributes
+        """
+        combined_edges = []
+        for edge in grouped_edges[node_name]:
+            # Find the inverse edge by looking in the group of the "to" node in the original edge.
+            inverse_edge = next((inv for inv in grouped_edges[edge["to"]] if inv["to"] == node_name), None)
+            up = edge["amount"]
+            down = 0
+            ratio = 0
+            log_ratio = 0
+            if inverse_edge:
+                down = inverse_edge["amount"]
+            if not(up == 0 and down == 0):
+                ratio = float(up) / (up + down)
+                log_ratio = log(up + 1) / (log(up + 1) + log(down + 1))
+            combined_edges.append({"from": node_name, "to": edge["to"], "amount_up": up, "amount_down": down,
+                                   "ratio": ratio, "log_ratio": log_ratio})
+        return combined_edges
+
+    @staticmethod
+    def group_elements(elements, attribute):
```

Comment:
This helper method contains some very generic dictionary functionality, I don't believe it should be in this module. Also, it's probably better to avoid using a `defaultdict` at all. I'm of the opinion it makes for unclear code, as usage of this dict in other places might have unexpected behavior for missing values.